.github/workflows/reusable_basic.yml

```diff
@@ -111,45 +118,34 @@ jobs:
             install_tbb: 'ON'
             disable_hwloc: 'OFF'
             link_hwloc_statically: 'ON'
-    runs-on: ${{matrix.os}}
-
     steps:
     - name: Checkout
       uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       with:
         fetch-depth: 0
 
-    - name: Install apt packages
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y clang cmake libnuma-dev lcov
-
-    - name: Install TBB apt package
-      if: matrix.install_tbb == 'ON'
-      run: |
-        sudo apt-get install -y libtbb-dev
-
     - name: Install oneAPI basekit
       if: matrix.compiler.cxx == 'icpx'
       run: |
+        sudo apt-get update
         sudo apt-get install -y gpg-agent wget
         wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor | sudo tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null
         echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
         sudo apt-get update
         sudo apt-get install -y intel-oneapi-ippcp-devel intel-oneapi-ipp-devel intel-oneapi-common-oneapi-vars intel-oneapi-compiler-dpcpp-cpp
-  
+
+    - name: Install TBB apt package
+      if: matrix.install_tbb == 'OFF'
+      run: |
+        sudo apt-get install -y libtbb-dev
+
     - name: Install g++-7
       if: matrix.compiler.cxx == 'g++-7'
```

Comment:
you ca get rid of this install step - `g++-7` is only tested on ubuntu-20.04 and it's already installed there